### PR TITLE
Feature/ui config jsonspec implementation

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Removes ```PluginEditCtrl```. Merges + simplifies (sort of) ```PluginEditCtrl``` & ```NodeConfigCtrl```.
- No more reference passing and editing the plugin properties. This PR connects the action to the store by invoking the actions.
- Makes returning a copy of anything from state in ```ConfigStore``` thereby preventing anybody who gets the data to directly modify things instead of going through action->dispatcher->store cycle.
- Handles both old and new config JSONs. So anybody who still uses 3.2 artifact (already published pipelines) will be able to use (backward compatibility)
- Modifies ```widget-container``` & all widgets to be compatible with both json specs
- Fixes ```server/aggregator``` to parse the json for syntax and semantic errors only for older spec.

GIST to modify old json to new json spec. [gist](https://gist.github.com/ajainarayanan/616b5a48b92ea9f0d85a)

### CODE REVIEW

File: cdap-ui/templates/oldToNewConfig.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  'cdap-etl-batch',
+  'cdap-etl-realtime',
+  'common'
+];
+
+var basePath = __dirname + '/';
+var fs = require('fs');
+
+path.forEach((folderName) => {
+  console.log('folderName: ', basePath + folderName);
```
Comment (elfenheart) [inline cdap-ui/templates/oldToNewConfig.js]: remove?

Comment (ajainarayanan) [inline cdap-ui/templates/oldToNewConfig.js]: Ah crap. No this file should not be in the repo. Its a throw away tool just for converting the old json to new one. All the jsons are going to be in hydrator-plugins. This is just for the transition. Will remove this file.

**Comment by elfenheart** [comment]
Positioning seems to be off between labels and field

screen shot 2015-12-03 at 5 28 54 pm

**Comment by elfenheart** [comment]
Nullable is not set in input schema

screen shot 2015-12-03 at 5 31 57 pm

**Comment by elfenheart** [comment]
also, can you rebase with latest develop?

**Comment by ajainarayanan** [comment]
@elfenheart Fixed the layout issue + rebased with latest develop. Could you give it a try?

**Comment by elfenheart** [comment]
@ajainarayanan the nullable is still an issue. Also found another issue, switching format to syslog, then change it to CSV, the schema seems to be empty instead of retaining the syslog's.

**Comment by ajainarayanan** [comment]
@elfenheart updated. 
bugfix

**Comment by ajainarayanan** [comment]
@elfenheart Have made one minor change to changing to different schema formats in stream source. The behavior right now is explained in this JIRA (https://issues.cask.co/browse/CDAP-4367). Will create a separate PR for this.

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**